django/db/utils.py

```diff
@@ -131,39 +132,34 @@ def load_backend(backend_name):
             raise
 
 
-class ConnectionDoesNotExist(Exception):
-    pass
-
+class ConnectionHandler(BaseConnectionHandler):
+    settings_name = 'DATABASES'
+    # Connections needs to still be an actual thread local, as it's truly
+    # thread-critical. Database backends should use @async_unsafe to protect
+    # their code from async contexts, but this will give those contexts
+    # separate connections in case it's needed as well. There's no cleanup
+    # after async contexts, though, so we don't allow that if we can help it.
+    thread_critical = True
 
-class ConnectionHandler:
-    def __init__(self, databases=None):
-        """
-        databases is an optional dictionary of database definitions (structured
-        like settings.DATABASES).
-        """
-        self._databases = databases
-        # Connections needs to still be an actual thread local, as it's truly
-        # thread-critical. Database backends should use @async_unsafe to protect
-        # their code from async contexts, but this will give those contexts
-        # separate connections in case it's needed as well. There's no cleanup
-        # after async contexts, though, so we don't allow that if we can help it.
-        self._connections = Local(thread_critical=True)
-
-    @cached_property
-    def databases(self):
-        if self._databases is None:
-            self._databases = settings.DATABASES
-        if self._databases == {}:
-            self._databases = {
+    def configure_settings(self, databases):
+        databases = super().configure_settings(databases)
+        if databases == {}:
+            databases.update({
                 DEFAULT_DB_ALIAS: {
                     'ENGINE': 'django.db.backends.dummy',
                 },
-            }
-        if DEFAULT_DB_ALIAS not in self._databases:
-            raise ImproperlyConfigured("You must define a '%s' database." % DEFAULT_DB_ALIAS)
-        if self._databases[DEFAULT_DB_ALIAS] == {}:
-            self._databases[DEFAULT_DB_ALIAS]['ENGINE'] = 'django.db.backends.dummy'
-        return self._databases
+            })
+        elif DEFAULT_DB_ALIAS not in databases:
+            raise ImproperlyConfigured(
+                f"You must define a '{DEFAULT_DB_ALIAS}' database."
+            )
+        if databases[DEFAULT_DB_ALIAS] == {}:
+            databases[DEFAULT_DB_ALIAS]['ENGINE'] = 'django.db.backends.dummy'
+        return databases
+
+    @cached_property
+    def databases(self):
+        return self.settings
```

Comment:
`self.settings` is already a `@cached_property`. If the cache needed clearing we'd need to then clear both `self.settings` and `self.databases`. So perhaps we should use `@property` here instead?

Anyway, ISTM that this is not something that is documented and we could just do away with `.databases` entirely and use `.settings` instead?

Reply:
I think we should leave `databases` for API stability, even if it's not documented.

Reply:
Ok. 👍